Elf/__init__.py

```diff
@@ -380,41 +436,254 @@ def _parseDynamic(self):
             else:
                 raise Exception('Platform not supported: %d' % (self.bits))
             dyn.vsParse(dynbytes)
+
             if dyn.d_tag in Elf32Dynamic.has_string:
                 name = self.getStrtabString(dyn.d_value, ".dynstr")
                 dyn.setName(name)
 
+            # don't add a second entry
+            if dyn not in self.dynamics:
+                logger.debug("dynamic: %r: 0x%x", dt_names.get(dyn.d_tag), dyn.d_value)
+                self.dynamics.append(dyn)
+
+            if dyn.d_tag == DT_NULL: # Represents the end
+                break
+            dynbytes = dynbytes[len(dyn):]
+
+    def _parseDynLinkInfo(self):
+        '''
+        Parse the Dynamics segment and populate both self.dynamics (legacy) and self.dyns
+        This must be run before most Dynamic-data accessors like getDynStrTabString(), 
+        getDynSymTabInfo(), etc..
+        '''
+        self.dyns = {}
+        dynbytes = self.getDynBytes()
+        if dynbytes is None:
+            return
+
+        while dynbytes:
+            if self.bits == 32:
+                dyn = Elf32Dynamic(bigend=self.bigend)
+            elif self.bits == 64:
+                dyn = Elf64Dynamic(bigend=self.bigend)
+            else:
+                raise Exception('Platform not supported: %d' % (self.bits))
+
+            dyn.vsParse(dynbytes)
+
+            # dump the tag/value pairs into the "dyns" dictionary.  if multiples, create a tuple
+            curdyn = self.dyns.get(dyn.d_tag)
+            if curdyn is not None:
+                self.dyns[dyn.d_tag] = (curdyn, dyn.d_value)
+            else:
+                self.dyns[dyn.d_tag] = dyn.d_value
+            logger.debug("dynamic: %r: 0x%x", dt_names.get(dyn.d_tag), dyn.d_value)
+
+            # FIXME: storing info in both dyns{} and dynamics[].  Is this a problem?
```

Comment:
Legacy.  Dynamics were originally stored in a list called dynamics, but i wanted to use a dict for ease of access throughout the ELF parsing process.  Since i don't know what someone else may have depending on the original, i created a second.  Perhaps we should deprecate the original one?